File: CS/S131432/Form1.cs
using DevExpress.XtraGrid.Views.Grid;
using System;
using System.Data;
using System.Drawing;
using System.Windows.Forms;

namespace S131432
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            productsBindingSource.DataSource = GetProductsDataTable();
        }
        DataTable GetProductsDataTable()
        {
            DataTable table = new DataTable();
            table.TableName = "Products";
            table.Columns.Add(new DataColumn("ProductID", typeof(int)));
            table.Columns.Add(new DataColumn("ProductName", typeof(string)));
            table.Columns.Add(new DataColumn("SupplierID", typeof(int)));
            table.Columns.Add(new DataColumn("CategoryID", typeof(string)));
            table.Columns.Add(new DataColumn("QuantityPerUnit", typeof(int)));
            for(int i = 0; i < 20; i++)
            {
                int index = i + 1;
                table.Rows.Add(index, "Product " + index, index * 2, "Category " + i % 8, index % 7);
            }
            return table;
        }
        private void OnGridViewRowStyle(object sender, RowStyleEventArgs e)
        {
            GridView gView = (GridView)sender;
            if(!gView.IsValidRowHandle(e.RowHandle)) return;
            int parent = gView.GetParentRowHandle(e.RowHandle);
            if(gView.IsGroupRow(parent))
            {
                for(int i = 0; i < gView.GetChildRowCount(parent); i++)
                    if(gView.GetChildRowHandle(parent, i) == e.RowHandle)
                        e.Appearance.BackColor = i % 2 == 0 ? Color.AliceBlue : Color.AntiqueWhite;
            }
            else e.Appearance.BackColor = e.RowHandle % 2 == 0 ? Color.AliceBlue : Color.AntiqueWhite;
        }
    }
}

File: VB/S131432/Form1.vb
Imports DevExpress.XtraGrid.Views.Grid
Imports System
Imports System.Data
Imports System.Drawing
Imports System.Windows.Forms

Namespace S131432
	Partial Public Class Form1
		Inherits Form

		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles Me.Load
			productsBindingSource.DataSource = GetProductsDataTable()
		End Sub
		Private Function GetProductsDataTable() As DataTable
			Dim table As New DataTable()
			table.TableName = "Products"
			table.Columns.Add(New DataColumn("ProductID", GetType(Integer)))
			table.Columns.Add(New DataColumn("ProductName", GetType(String)))
			table.Columns.Add(New DataColumn("SupplierID", GetType(Integer)))
			table.Columns.Add(New DataColumn("CategoryID", GetType(String)))
			table.Columns.Add(New DataColumn("QuantityPerUnit", GetType(Integer)))
			For i As Integer = 0 To 19
				Dim index As Integer = i + 1
				table.Rows.Add(index, "Product " & index, index * 2, "Category " & i Mod 8, index Mod 7)
			Next i
			Return table
		End Function
		Private Sub OnGridViewRowStyle(ByVal sender As Object, ByVal e As RowStyleEventArgs) Handles gridView1.RowStyle
			Dim gView As GridView = DirectCast(sender, GridView)
			If Not gView.IsValidRowHandle(e.RowHandle) Then
				Return
			End If
'INSTANT VB NOTE: The variable parent was renamed since Visual Basic does not handle local variables named the same as class members well:
			Dim parent_Renamed As Integer = gView.GetParentRowHandle(e.RowHandle)
			If gView.IsGroupRow(parent_Renamed) Then
				Dim i As Integer = 0
				Do While i < gView.GetChildRowCount(parent_Renamed)
					If gView.GetChildRowHandle(parent_Renamed, i) = e.RowHandle Then
						e.Appearance.BackColor = If(i Mod 2 = 0, Color.AliceBlue, Color.AntiqueWhite)
					End If
					i += 1
				Loop
			Else
				e.Appearance.BackColor = If(e.RowHandle Mod 2 = 0, Color.AliceBlue, Color.AntiqueWhite)
			End If
		End Sub
	End Class
End Namespace